diff --git a/RNFBApp.podspec b/RNFBApp.podspec
index 0f8ca1d42cbb173f55062c49560356ae09306b6b..7ca6a54dfd28f85efc6de8be352cee7300c5b407 100644
--- a/RNFBApp.podspec
+++ b/RNFBApp.podspec
@@ -41,6 +41,7 @@ Pod::Spec.new do |s|
   # App must define modules for static framework integration of other packages to work
   s.pod_target_xcconfig = {
     "DEFINES_MODULE" => "YES",
+    "CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES" => "YES",
   }

   # React Native dependencies
@@ -50,6 +51,10 @@ Pod::Spec.new do |s|
     s.dependency "React-Core"
   end

+  if defined?(add_rncore_dependency)
+    add_rncore_dependency(s)
+  end
+
   if (ENV.include?('FIREBASE_SDK_VERSION'))
     Pod::UI.puts "#{s.name}: Found Firebase SDK version in environment '#{ENV['FIREBASE_SDK_VERSION']}'"
     $FirebaseSDKVersion = ENV['FIREBASE_SDK_VERSION']
diff --git a/ios/RNFBApp/RNFBAppModule.m b/ios/RNFBApp/RNFBAppModule.m
index f679daad1b3586c534dba0ba40c5010a18a9d1ed..822e31fca7633c46f9a0aa406472fefe2aef5c7a 100644
--- a/ios/RNFBApp/RNFBAppModule.m
+++ b/ios/RNFBApp/RNFBAppModule.m
@@ -16,6 +16,7 @@
  */

 #import <Firebase/Firebase.h>
+#import <React/RCTBridgeModule.h>
 #import <React/RCTUtils.h>

 #import "RNFBAppModule.h"
diff --git a/ios/RNFBApp/RNFBUtilsModule.m b/ios/RNFBApp/RNFBUtilsModule.m
index 7d283311095b94a599567ffa20731f58f20e1c33..dfb35e4270987841d02f638a0419f764dba50c6a 100644
--- a/ios/RNFBApp/RNFBUtilsModule.m
+++ b/ios/RNFBApp/RNFBUtilsModule.m
@@ -15,6 +15,7 @@
  *
  */

+#import <React/RCTBridgeModule.h>
 #import <React/RCTUtils.h>

 #import "RNFBApp/RNFBSharedUtils.h"
