diff --git a/RNFBAuth.podspec b/RNFBAuth.podspec
index 8e9b9d50653f06c8b9027056b6a54f4b2b2441e6..81ca4275191ab850a3c3026bce8e80c165c2ac1e 100644
--- a/RNFBAuth.podspec
+++ b/RNFBAuth.podspec
@@ -38,6 +38,14 @@ Pod::Spec.new do |s|
     s.dependency "React-Core"
   end

+  if defined?(add_rncore_dependency)
+    add_rncore_dependency(s)
+  end
+
+  s.pod_target_xcconfig = {
+    "CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES" => "YES",
+  }
+
   if defined?($FirebaseSDKVersion)
     Pod::UI.puts "#{s.name}: Using user specified Firebase SDK version '#{$FirebaseSDKVersion}'"
     firebase_sdk_version = $FirebaseSDKVersion
diff --git a/ios/RNFBAuth/RNFBAuthModule.m b/ios/RNFBAuth/RNFBAuthModule.m
index 73b91b9a4b63c8787b99bf3126110ef22c62c49b..7bf0be37b0313baa1f1087f70aff76956b913915 100644
--- a/ios/RNFBAuth/RNFBAuthModule.m
+++ b/ios/RNFBAuth/RNFBAuthModule.m
@@ -16,6 +16,7 @@
  */

 #import <Firebase/Firebase.h>
+#import <React/RCTBridgeModule.h>
 #import <React/RCTUtils.h>

 #import "RNFBApp/RCTConvert+FIRApp.h"
