diff --git a/RNFBFirestore.podspec b/RNFBFirestore.podspec
index a153aa395d9bea6d9b61330b994d80e6b7b01352..d407854a7caeca20db1b79a5a35643caca22ad3a 100644
--- a/RNFBFirestore.podspec
+++ b/RNFBFirestore.podspec
@@ -39,6 +39,14 @@ Pod::Spec.new do |s|
     s.dependency "React-Core"
   end

+  if defined?(add_rncore_dependency)
+    add_rncore_dependency(s)
+  end
+
+  s.pod_target_xcconfig = {
+    "CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES" => "YES",
+  }
+
   if defined?($FirebaseSDKVersion)
     Pod::UI.puts "#{s.name}: Using user specified Firebase SDK version '#{$FirebaseSDKVersion}'"
     firebase_sdk_version = $FirebaseSDKVersion
diff --git a/ios/RNFBFirestore/RNFBFirestoreCollectionModule.h b/ios/RNFBFirestore/RNFBFirestoreCollectionModule.h
index c9db600cf9ef53ce9225774b758b86c9766e92b7..d570527437d0a8614b4b7f5fe11218a3f77401ab 100644
--- a/ios/RNFBFirestore/RNFBFirestoreCollectionModule.h
+++ b/ios/RNFBFirestore/RNFBFirestoreCollectionModule.h
@@ -17,9 +17,9 @@

 #import <Firebase/Firebase.h>
 #import <Foundation/Foundation.h>
+#import <React/RCTBridgeModule.h>
 #import <RNFBApp/RNFBSharedUtils.h>
 #import <RNFBFirestoreQuery.h>
-#import <React/RCTBridgeModule.h>
 #import "RNFBFirestoreCommon.h"
 #import "RNFBFirestoreSerialize.h"

diff --git a/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m b/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m
index dd21cf1896b35ca3ffa4e514319c167a3612fd77..646f15bfa1afc54badafcda1b9c68bfb6c0babdb 100644
--- a/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m
+++ b/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m
@@ -15,6 +15,8 @@
  *
  */

+#import <React/RCTBridgeModule.h>
+#import <React/RCTUtils.h>
 #import <RNFBApp/RNFBRCTEventEmitter.h>
 #if __has_include(<RNFBFirestore/RNFBFirestore-Swift.h>)
 // This import will work in situations where `use_frameworks!` is in use
@@ -27,8 +29,6 @@
 // directory. See firebase-ios-sdk#12611 for more context.
 #import "RNFBFirestore-Swift.h"
 #endif
-#import <React/RCTUtils.h>
-
 #import "RNFBFirestoreCollectionModule.h"
 #import "RNFBFirestoreCommon.h"

diff --git a/ios/RNFBFirestore/RNFBFirestoreDocumentModule.h b/ios/RNFBFirestore/RNFBFirestoreDocumentModule.h
index 8a314bdce1f8dccba27ab7436f7fd5f3fb11ed3f..8b0b6fa91f13d51e5ee7776e556eb13492da5a89 100644
--- a/ios/RNFBFirestore/RNFBFirestoreDocumentModule.h
+++ b/ios/RNFBFirestore/RNFBFirestoreDocumentModule.h
@@ -17,8 +17,8 @@

 #import <Firebase/Firebase.h>
 #import <Foundation/Foundation.h>
-#import <RNFBApp/RNFBSharedUtils.h>
 #import <React/RCTBridgeModule.h>
+#import <RNFBApp/RNFBSharedUtils.h>
 #import "RNFBFirestoreCommon.h"
 #import "RNFBFirestoreSerialize.h"

diff --git a/ios/RNFBFirestore/RNFBFirestoreDocumentModule.m b/ios/RNFBFirestore/RNFBFirestoreDocumentModule.m
index 6e51e42d12b0ea913a7a9da4dd366d614efd3b03..76207814974a6c7b5cb283c8b4a4c02b1e63d62c 100644
--- a/ios/RNFBFirestore/RNFBFirestoreDocumentModule.m
+++ b/ios/RNFBFirestore/RNFBFirestoreDocumentModule.m
@@ -15,8 +15,9 @@
  *
  */

-#import <RNFBApp/RNFBRCTEventEmitter.h>
+#import <React/RCTBridgeModule.h>
 #import <React/RCTUtils.h>
+#import <RNFBApp/RNFBRCTEventEmitter.h>

 #import "RNFBFirestoreDocumentModule.h"

diff --git a/ios/RNFBFirestore/RNFBFirestoreModule.h b/ios/RNFBFirestore/RNFBFirestoreModule.h
index a989534e02503d16f7d2289acebecab888a064d6..d2f3016b2a0455950b0583490242c5e983c9e386 100644
--- a/ios/RNFBFirestore/RNFBFirestoreModule.h
+++ b/ios/RNFBFirestore/RNFBFirestoreModule.h
@@ -17,8 +17,8 @@

 #import <Firebase/Firebase.h>
 #import <Foundation/Foundation.h>
-#import <RNFBApp/RNFBSharedUtils.h>
 #import <React/RCTBridgeModule.h>
+#import <RNFBApp/RNFBSharedUtils.h>

 @interface RNFBFirestoreModule : NSObject <RCTBridgeModule>

diff --git a/ios/RNFBFirestore/RNFBFirestoreModule.m b/ios/RNFBFirestore/RNFBFirestoreModule.m
index 5b9c7a63c8ce3ca4d46c80b3e9c3ca824657e419..449ffd69d2c4a203d65994f6d3f6928651a2b3e4 100644
--- a/ios/RNFBFirestore/RNFBFirestoreModule.m
+++ b/ios/RNFBFirestore/RNFBFirestoreModule.m
@@ -15,9 +15,10 @@
  *
  */

+#import <React/RCTBridgeModule.h>
+#import <React/RCTUtils.h>
 #import "RNFBFirestoreModule.h"
 #import <RNFBApp/RNFBRCTEventEmitter.h>
-#import <React/RCTUtils.h>
 #import "FirebaseFirestoreInternal/FIRPersistentCacheIndexManager.h"
 #import "RNFBFirestoreCommon.h"
 #import "RNFBPreferences.h"
diff --git a/ios/RNFBFirestore/RNFBFirestoreTransactionModule.h b/ios/RNFBFirestore/RNFBFirestoreTransactionModule.h
index 7fb14bb3576c982e69017fcd7537cfeb29632d36..e9f03278c75eeb0aafe99e78a9d68defe136cbe1 100644
--- a/ios/RNFBFirestore/RNFBFirestoreTransactionModule.h
+++ b/ios/RNFBFirestore/RNFBFirestoreTransactionModule.h
@@ -17,9 +17,9 @@

 #import <Firebase/Firebase.h>
 #import <Foundation/Foundation.h>
+#import <React/RCTBridgeModule.h>
 #import <RNFBApp/RNFBSharedUtils.h>
 #import <RNFBFirestoreQuery.h>
-#import <React/RCTBridgeModule.h>
 #import "RNFBFirestoreCommon.h"
 #import "RNFBFirestoreSerialize.h"

diff --git a/ios/RNFBFirestore/RNFBFirestoreTransactionModule.m b/ios/RNFBFirestore/RNFBFirestoreTransactionModule.m
index bf3dad6dda2f016fb8dc4fa86b0d2855c234001d..be0ab40f87ec53377826b9db9cf3fa2d2713b9a0 100644
--- a/ios/RNFBFirestore/RNFBFirestoreTransactionModule.m
+++ b/ios/RNFBFirestore/RNFBFirestoreTransactionModule.m
@@ -15,7 +15,8 @@
  *
  */

-#import <RNFBApp/RNFBRCTEventEmitter.h>
+#import <React/RCTBridgeModule.h>
 #import <React/RCTUtils.h>
+#import <RNFBApp/RNFBRCTEventEmitter.h>

 #import "RNFBFirestoreTransactionModule.h"
